File: Java Script - Introduction to Programming/3.2. Simple Conditions - Exam Problems/04. Harvest.js
function harvest(arg) {
  let [vineyard, grapePerSquare, wineNeed, workers] = [...arg.map(Number)]
  let totalGrape = vineyard * grapePerSquare;
  let totalWine = totalGrape * (40.0 / 100) / 2.5

  if (totalWine >= wineNeed) {
    console.log(`Good harvest this year! Total wine: ${Math.floor(totalWine)} liters.`)
    console.log(`${Math.ceil(totalWine-wineNeed)} liters left -> ${Math.ceil((totalWine-wineNeed)/workers)} liters per person.`)
  } else {
    console.log(`It will be a tough winter! More ${Math.floor(Math.abs(totalWine-wineNeed))} liters wine needed.`)
  }
}

//harvest([650, 2, 175, 3])

File: Pyrhon - Introduction to Programming/3.2. Simple Conditions - Exam Problems/04. Harvest.py
from math import floor,ceil

vineyard, grape_per_square, wine_need, workers = [float(input()) for _ in range(4)]
total_grape = vineyard * grape_per_square
total_wine = total_grape * (40.0 / 100) / 2.5

if total_wine >= wine_need:
    print(f'Good harvest this year! Total wine: {floor(total_wine)} liters.')
    print(f'{ceil(total_wine - wine_need)} liters left -> {ceil((total_wine - wine_need) / workers)} liters per person.')
else:
    print(f'It will be a tough winter! More {floor(abs(total_wine - wine_need))} liters wine needed.')


'''
650
2
175
3
'''
